monkestation/code/modules/blacksmithing/anvil.dm

```diff
@@ -0,0 +1,325 @@
+	var/steptime = 50
+	playsound(src, 'sound/effects/clang.ogg',40, 2)
+	if(!do_after(user, steptime, target = src))
+		return FALSE
+	switch(stepdone)
+		if("weak hit")
+			currentsteps += 1
+			outrightfailchance += 5
+			currentquality += 1
+		if("strong hit")
+			currentsteps += 2
+			outrightfailchance += 9.5
+			currentquality += 2
+		if("heavy hit")
+			currentsteps += 3
+			outrightfailchance += 12.5
+			currentquality += 3
+		if("fold")
+			stepsdone += "f"
+			currentsteps += 1
+			currentquality -= 1
+		if("draw")
+			stepsdone += "d"
+			currentsteps += 1
+			currentquality -= 1
+		if("shrink")
+			stepsdone += "s"
+			currentsteps += 1
+			currentquality -= 1
+		if("bend")
+			stepsdone += "b"
+			currentsteps += 1
+			currentquality -= 1
+		if("punch")
+			stepsdone += "p"
+			currentsteps += 1
+			currentquality -= 1
+		if("upset")
+			stepsdone += "u"
+			currentsteps += 1
+			currentquality -= 1
+	user.visible_message("<span class='notice'>[user] works the metal on the anvil with their hammer with a loud clang!</span>", \
+						"<span class='notice'>You [stepdone] the metal with a loud clang!</span>")
+	playsound(src, 'sound/effects/clang.ogg',40, 2)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/playsound, src, 'sound/effects/clang.ogg', 40, 2), 15)
+	if(length(stepsdone) >= 3)
+		tryfinish(user)
+
+/obj/structure/anvil/proc/tryfinish(mob/user)
+	var/artifactchance = 0
+	if(!artifactrolled)
+		artifactchance = (1+ rand(1, 25) / 2500)
+		artifactrolled = TRUE
+	var/artifact = max(prob(artifactchance), debug)
+	var/finalfailchance = (outrightfailchance * workpiece_material[1].fail_multipler)
+	finalfailchance = max(0, finalfailchance / (1 + anvilquality)) //lv 2 gives 20% less to fail, 3 30%, etc
+	if((currentsteps > 10 || (rng && prob(finalfailchance))) && !artifact)
+		to_chat(user, "<span class='warning'?You overwork the metal, causing it to turn into useless slag!</span>")
+		var/turf/Turf = get_turf(user)
+		workpiece_state = FALSE
+		new /obj/item/stack/ore/slag(Turf)
+		currentquality = anvilquality
+		stepsdone = ""
+		currentsteps = 0
+		outrightfailchance = 1
+		artifactrolled = FALSE
+	for(var/recipe in smithrecipes)
+		if(recipe == stepsdone)
+			var/turf/Turf = get_turf(user)
+			var/obj/item/smithing/create = smithrecipes[stepsdone]
+			var/obj/item/smithing/finisheditem = new create(Turf)
+			to_chat(user, "You finish your [finisheditem]!")
+			if(artifact)
+				to_chat(user, "It is an artifact, a creation whose legacy shall live on forevermore.") //todo: SSblackbox
+				currentquality = max(currentquality, 2)
+				finisheditem.quality = currentquality * 3//this is insane i know it's 1/2500 for most of the time and 0.8% at best
+				finisheditem.artifact = TRUE
+			else
+				finisheditem.quality = min(currentquality, itemqualitymax)
+			switch(finisheditem.quality)
+				if(-1000 to -8)
+					finisheditem.desc =  "It looks to be the most awfully made object you've ever seen."
+				if(-8)
```

Comment:
This will never be reached, as the previous switch statement will always catch it